solr/core/src/java/org/apache/solr/highlight/DefaultSolrHighlighter.java

```diff
@@ -570,7 +571,8 @@ protected void flatten(
       fieldHighlights =
           doHighlightingByFastVectorHighlighter(doc, docId, schemaField, fvhContainer, reader, req);
     } else { // standard/default highlighter
-      fieldHighlights = doHighlightingByHighlighter(doc, docId, schemaField, query, reader, req);
+      fieldHighlights =
+          doHighlightingByHighlighter(doc, docId, schemaField, query, reader.termVectors(), req);
```

Comment:
`reader.termVectors()` is called in a loop here, I'll push a commit to try and replace that.